### PR TITLE
Test project details module

### PR DESCRIPTION
**Pull request by Shrawak**
Ticket Link:
[ClickUp Ticket Header](https://app.clickup.com/t/864evfgkw)
Description:
* Test should render circularprogress when loading
* Test should render project detail
* Test should render project detail page with error
* Test should not delete project as there are project assignments
* Test should add project assignment
* Test should not add project assignment as there are no employees
* Test should delete project
* Test should delete project Assignment
* Test should not delete project assignment as there are no project assignments

Screenshots/GIFs:
image

image

image

image

image

image

image

image

image

Testing Strategy:
yarn test

### CODE REVIEW

File: src/containers/Project/test/projectDetails.test.tsx [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockedUsedNavigate,
+}));
+
+jest.mock('@/hooks/useCurrentUser');
+jest.mock('@/hooks/useProjectAssignments');
+jest.mock('@/hooks/useProjectById');
+jest.mock('@/hooks/useAllEmployee');
+jest.mock('@/hooks/useDeleteProjectAssignment');
```
Comment (opencafe-eng) [inline src/containers/Project/test/projectDetails.test.tsx]: why do we have hooks for different actions. It should have been inside useProjectAssignment. 
useDelete, useAdd, useEdit everything should be inside useProjectAssignment.

File: src/containers/Project/test/projectDetails.test.tsx [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    act(() => {
+      employeeAutoComplete.focus();
+      // type employee Name
+      fireEvent.change(employeeAutoComplete, {
+        target: { value: 'Shreyam Pokharel' },
+      });
+    });
+
+    // navigate to the first item in the autocomplete box
+    fireEvent.keyDown(employeeAutoComplete, { key: 'ArrowDown' });
```
Comment (opencafe-eng) [inline src/containers/Project/test/projectDetails.test.tsx]: userEvent instead of fireevent

**Review by opencafe-eng — CHANGES_REQUESTED** [review]

File: src/hooks/useProject/index.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { useQuery, useMutation } from '@tanstack/react-query';
+import {
+  fetchProject,
+  addProject,
+  deleteProject,
+  fetchProjectById,
+} from './request';
+
+const useProject = (projectId?: string, page?: number, limit?: number) => {
```
Comment (duwalanise) [inline src/hooks/useProject/index.ts]: use object for props here since all of the args are optional using object will allow more flexible call.

**Review by duwalanise — APPROVED** [review]

File: src/hooks/useProject/index.ts (hunk 3 above)
Comment (Shrawak) [reply src/hooks/useProject/index.ts]: Like this?
interface IProject {
  projectId?: string;
  page?: number;
  limit?: number;
}
1)
const useProject = ({ projectId, page, limit }: IProject) => {
2)
const useProject = (props: IProject) => {

Which one to do first one or second one or any other approach?

**Pull request merged**